src/components/general/message.tsx

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable react/require-default-props */
+import { useAtomValue } from 'jotai';
+import { Suspense } from 'react';
+import {
+  GUI_LOCALIZATION_ATOM,
+  MessageType,
+  MessageResolverAtom,
+} from '../../localization/localization';
+
+interface MessageProps {
+  message: MessageType;
+  alternativeSource?: MessageResolverAtom;
+}
+
+// usage for cases where node not directly applicable, but still inside React context
+export function useMessage(alternativeSource?: MessageResolverAtom) {
+  return useAtomValue(alternativeSource ?? GUI_LOCALIZATION_ATOM);
+}
+
+function InnerMessage(props: MessageProps) {
+  const { message, alternativeSource } = props;
+  const localize = useMessage(alternativeSource);
+  return localize(message);
+}
+
+export default function Message(props: Partial<MessageProps>) {
+  const { message, alternativeSource } = props;
+  if (!message) {
+    return null;
+  }
+
+  return (
+    <Suspense>
+      <InnerMessage message={message} alternativeSource={alternativeSource} />
```

Comment:
![image](https://github.com/user-attachments/assets/b4c4be95-b230-4b95-980a-309014cd19c2)
Every time I switch tabs I get this error in the console. Why is it shown every time? Is the locale message reparsed every time on tab switch? I expected some kind of caching behavior I guess. Do you know why this occurs?

Reply:
Added a simple `memo` for the Message-Component.  
The direct usage of `useMessage` has no real opportunity for easy memoization, but adding it around the components already helps with non-dynamic text.